packages/service/tests/unit/test_discovery_client.py

```diff
@@ -128,13 +128,16 @@ def test___service_registered___resolve_service___sends_request(
     )
 
     service_location = discovery_client.resolve_service(
-        _TEST_SERVICE_INFO.provided_interfaces[0], _TEST_SERVICE_INFO.service_class
+        provided_interface=_TEST_SERVICE_INFO.provided_interfaces[0],
+        service_class=_TEST_SERVICE_INFO.service_class,
+        version=_TEST_SERVICE_INFO.versions[0],
```

Comment:
You are adding a new optional parameter, so you should keep the existing test cases to ensure you didn't break existing usage. Add a new test case that exercises the new parameter.